src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,40 @@
 package mate.academy.lib;
 
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
+    private static final Map<Class<?>, Class<?>> interfaceToImplementation = new HashMap<>();
     private static final Injector injector = new Injector();
 
+    static {
+        interfaceToImplementation.put(ProductService.class, ProductServiceImpl.class);
+        interfaceToImplementation.put(ProductParser.class, ProductParserImpl.class);
+        interfaceToImplementation.put(FileReaderService.class, FileReaderServiceImpl.class);
+    }
+
+    private Injector() {
+    }
+
     public static Injector getInjector() {
         return injector;
     }
 
-    public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+    public <T> T getInstance(Class<T> interfaceClazz) {
+        Class<?> implClass = interfaceToImplementation.get(interfaceClazz);
```

Comment:
This is not full implementation os Injector

The convention is to use Component on the classes that are used for instance creation (usually these are interface implementations) and Inject to mark fields that need to be initialized. If the Component annotation is missing above the class, we shouldn't be able to create an instance of this class in Injector, and we should throw an exception

Don't forget about annotations checks in your Injector class.
Let's check instances map before new instance creation.